File: app/src/main/java/com/maksk993/cryptoportfolio/domain/usecases/GetPricesFromCoinMarketCap.java
package com.maksk993.cryptoportfolio.domain.usecases;

import com.maksk993.cryptoportfolio.domain.repository.CryptoRepository;
import com.maksk993.cryptoportfolio.domain.models.DataReceivedCallBack;

public class GetPricesFromCoinMarketCap {
    private final CryptoRepository repository;

    public GetPricesFromCoinMarketCap(CryptoRepository repository){
        this.repository = repository;
    }
    public void execute(DataReceivedCallBack callBack){
        repository.getPrices(callBack);
    }
}

File: app/src/main/java/com/maksk993/cryptoportfolio/domain/usecases/GetPricesFromCoinMarketCap.kt
package com.maksk993.cryptoportfolio.domain.usecases

import com.maksk993.cryptoportfolio.domain.models.Asset
import com.maksk993.cryptoportfolio.domain.repository.CryptoRepository
import kotlinx.coroutines.flow.Flow

class GetPricesFromCoinMarketCap(private val repository: CryptoRepository) {
    suspend fun execute(): Flow<Asset> = repository.getPrices()
}